### PR TITLE
feat: init ktx2 loader

### PR DESCRIPTION
**Pull request by RenaudRohlinger**
### What
New hook `useKTX2` for KTX2 compression textures.

Demo:
https://drei-git-fork-utsuboco-feature-ktx2loader-pmndrs.vercel.app/?path=/story/loaders-ktx2--use-ktx-2-scene-st

### Checklist

- [x] Documentation updated
- [x] Storybook entry added
- [x] Ready to be merged

### CODE REVIEW

File: src/core/useKTX2.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { IsObject } from './useTexture'
+
+export function useKTX2<Url extends string[] | string | Record<string, string>>(
+  input: Url,
+  basisPath?: string
+): Url extends any[] ? Texture[] : Url extends object ? { [key in keyof Url]: Texture } : Texture {
+  const gl = useThree((state) => state.gl)
+  const textures = useLoader(KTX2Loader, IsObject(input) ? Object.values(input) : (input as any), (loader: any) => {
+    loader.detectSupport(gl)
+    loader.setTranscoderPath(basisPath || '/basis/')
```
Comment (gsimone) [inline src/core/useKTX2.tsx:14]:
```suggestion
  basisPath?: string =  '/basis/'
): Url extends any[] ? Texture[] : Url extends object ? { [key in keyof Url]: Texture } : Texture {
  const gl = useThree((state) => state.gl)
  const textures = useLoader(KTX2Loader, IsObject(input) ? Object.values(input) : (input as any), (loader: any) => {
    loader.detectSupport(gl)
    loader.setTranscoderPath(basisPath)
```

tiny change moving the default to the args instead

File: src/core/useKTX2.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    return keyed
+  } else {
+    return textures as any
+  }
+}
+
+useKTX2.preload = (url: string extends any[] ? string[] : string, basisPath: string) =>
+  useLoader.preload(KTX2Loader, url, (loader: any) => {
+    loader.setTranscoderPath(basisPath || '/basis/')
+  })
```
Comment (gsimone) [inline src/core/useKTX2.tsx:37]: I'd also change this to be a default to the arg instead of ||

**Review by gsimone — APPROVED** [review]

**Comment by gsimone** [comment]
Also from discord
> you could use the drei-assets repo and use jsdelivr/cdnjs [to load the binaries]

**Comment by RenaudRohlinger** [comment]
@gsimone Can merge once https://github.com/pmndrs/drei-assets/pull/3 is merged and storybook is tested and working

**Comment by RenaudRohlinger** [comment]
It works
https://drei-git-fork-utsuboco-feature-ktx2loader-pmndrs.vercel.app/?path=/story/loaders-ktx2--use-ktx-2-scene-st

**Comment by rslndzh** [comment]
Is there any way to load GLTF with KTX textures via useGLTF ?

**Pull request merged**